internal/clients/diffreviewer/circleci/circleci_service.go

```diff
@@ -0,0 +1,106 @@
+package circleci
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/diffreviewer"
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/gitdiff"
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/logger"
+	circlelogger "github.com/nightfallai/nightfall_code_scanner/internal/clients/logger/circle_logger"
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/nightfall"
+	"github.com/nightfallai/nightfall_code_scanner/internal/interfaces/gitdiffintf"
+	"github.com/nightfallai/nightfall_code_scanner/internal/nightfallconfig"
+)
+
+const (
+	WorkspacePathEnvVar          = "GITHUB_WORKSPACE"
+	NightfallAPIKeyEnvVar        = "NIGHTFALL_API_KEY"
+	CircleRepoNameEnvVar         = "CIRCLE_PROJECT_REPONAME"
+	CircleOwnerNameEnvVar        = "CIRCLE_PROJECT_USERNAME"
+	CircleCurrentCommitShaEnvVar = "CIRCLE_SHA1"
+	CircleBeforeCommitEnvVar     = "EVENT_BEFORE"
+	CirclePullRequestUrlEnvVar   = "CIRCLE_PULL_REQUEST"
+)
+
+// Service contains the github client that makes Github api calls
+type Service struct {
+	Logger  logger.Logger
+	GitDiff gitdiffintf.GitDiff
+}
+
+// NewCircleCiService creates a new CircleCi service
+func NewCircleCiService() diffreviewer.DiffReviewer {
+	return &Service{
+		Logger: circlelogger.NewDefaultCircleLogger(),
+	}
+}
+
+// GetLogger gets the github service logger
+func (s *Service) GetLogger() logger.Logger {
+	return s.Logger
+}
+
+// LoadConfig gets all config values from files or environment and creates a config
+func (s *Service) LoadConfig(nightfallConfigFileName string) (*nightfallconfig.Config, error) {
+	s.Logger.Debug("Loading configuration")
+	workspacePath, ok := os.LookupEnv(WorkspacePathEnvVar)
+	if !ok && workspacePath != "" {
```

Comment:
sorry I was writing quickly - I think we actually want to test that the workspacePath IS empty here (==). Is there any situation in which this can be empty?

Reply:
shoot my bad haha - and no I don't think so since I'm manually setting it into the environment in the circle orb. I think we should still check that it's not empty however in case it's not there